pkg/git/local_repository.go

```diff
@@ -326,35 +327,84 @@ type PullResult struct {
 	IsUpdated bool
 }
 
-func (localRepo *LocalRepository) Pull(auth AuthMethod) (PullResult, error) {
+func (localRepo *LocalRepository) Pull(auth AuthMethod) (*PullResult, error) {
 	var gitAuth transport.AuthMethod
 	if auth != nil {
 		gitAuth = auth.toGitAuthMethod()
 	}
+
 	var err error
+	originType := unknownOriginType
+	if !localRepo.DryRun {
+		originType, err = localRepo.originType()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	goGit := originType == httpsOriginType
+
 	log.Debug().
 		Str("repo", localRepo.Path()).
+		Bool("go_git", goGit).
 		Bool("dry_run", localRepo.DryRun).
 		Msg("git: pull")
 	if !localRepo.DryRun {
-		err = localRepo.worktree.Pull(&git.PullOptions{
-			RemoteName: OriginRemote,
-			Auth:       gitAuth,
-		})
-	}
-	if err != nil && !errors.Is(err, git.NoErrAlreadyUpToDate) {
-		return PullResult{}, err
+		if goGit {
+			err = localRepo.worktree.Pull(&git.PullOptions{
+				RemoteName: OriginRemote,
+				Auth:       gitAuth,
+			})
+			if err != nil && !errors.Is(err, git.NoErrAlreadyUpToDate) {
+				return nil, err
+			}
+		} else {
+			if stdout, stderr, err := shell.RunCommand(localRepo.Path(), "git", "pull", OriginRemote); err != nil {
```

Comment:
Why not using it git binary everytime?
I think it should simplify things

Reply:
If we only use the git binary then the github token in config will not work unless we configure git for it.

Reply:
How does it affect the token?
I don't think I understand
Or do you mean that in case remote protocol is https, then it won't work, because the token won't be injected in url?